### ISSUE TITLE
correcting search item

### ISSUE THREAD

**Issue by marzieh67** [marzieh67/ReDI_Frontend_FinalProject#6]
I do not know where I am making a mistake in this search button. have tried a lot on my local file but still not working :confused: 
should try again and create a new PR afterwards

**Comment by golovnevs** [comment]
Good job! I just made couple things to make it work :) Repeat my steps and feel free to ask me, if you have a problem. Also we can make one to one session, if you want.

1. remove variables lat and lon - you don't need it. You can call api only with the city name and key. 

http://api.openweathermap.org/data/2.5/weather?q=munich&appid=c71eb334b9f16ce9ac215af9464c2b86

2. change <input type="button"> to <button type="submit"> (input doesn't causes refresh, submit refreshes the page)
3. add id for the "city input" form and "search" button
4.create variable to store city name, take it from input form by id (let cityName=document.getElementById("cityInput").value ), and put it to the API request link (like you did it with the key ${cityName}) 
5. create event listener for the search button :) 
6. Enjoy your work :)

**Comment by marzieh67** [comment]
@golovnevs Thanks a lot Sergey. actually I tried with city name as well, but was not successful. I appreciate your suggestions and will update it :+1:

**Comment by marzieh67** [comment]
> Good job! I just made couple things to make it work :) Repeat my steps and feel free to ask me, if you have a problem. Also we can make one to one session, if you want.
> 
> 1. remove variables lat and lon - you don't need it. You can call api only with the city name and key.
> 
> http://api.openweathermap.org/data/2.5/weather?q=munich&appid=c71eb334b9f16ce9ac215af9464c2b86
> 
> For temperature in Celsius use &units=metric at the end of the link
> 
> http://api.openweathermap.org/data/2.5/weather?q=munich&appid=c71eb334b9f16ce9ac215af9464c2b86&units=metric
> 
> 1. change < button type="submit" > to  < input type="button" > (input doesn't causes refresh, submit refreshes the page)
> 2. add id for the "city input" form and "search" button
>    4.create variable to store city name, take it from input form by id (let cityName=document.getElementById("cityInput").value ), and put it to the API request link (like you did it with the key ${cityName})
> 3. create event listener for the search button :)
> 4. Enjoy your work :)
@golo

**Comment by marzieh67** [comment]
```javascript
const iconElement = document.querySelector(".weather-icon");
const tempElement = document.querySelector(".temperature-value p");
const tempFeelsElement = document.querySelector(".temperature-feels-value p");
const tempMinElement = document.querySelector(".temperature-min-value p");
const tempMaxElement = document.querySelector(".temperature-max-value p");
const descElement = document.querySelector(".temperature-description p");
const locationElement = document.querySelector(".location p");
const humidityElement = document.querySelector(".humidity p");

//App data

const weather = {};

//GET WEATHER FROM API Provider
const key = "c71eb334b9f16ce9ac215af9464c2b86";
let city_name = document.getElementById("searchbar").value;
console.log(city_name);

document.getElementById('citysearch').addEventListener('submit', getWeather);


//getWeather();

function getWeather() {
    //const api= `http://api.openweathermap.org/data/2.5/weather?q=London,uk&appid=${key}`;
    const api = `http://api.openweathermap.org/data/2.5/weather?q=${city_name}&appid=${key}&units=metric`;
    console.log(api);
    fetch(api).then(function (response) {
        let data = response.json();
        return data;
        console.log(data);
    })
        .then(function (data) {
            weather.temperature = Math.floor(data.main.temp);
            weather.temperature_FeelsLike = Math.floor(data.main.feels_like);
            weather.temperatureMin = Math.floor(data.main.temp_min);
            weather.temperatureMax = Math.floor(data.main.temp_max);
            weather.humidity = data.main.humidity;
            weather.description = data.weather[0].description;
            weather.iconId = data.weather[0].icon;
            weather.city = data.name;
            weather.country = data.sys.country;
            console.log(weather.humidity);
            console.log(weather.description);
        })
        .then(function () {
            displayWeather();
        });

}

function displayWeather() {
    iconElement.innerHTML = `<img src="icons/${weather.iconId}.png"/>`;
    tempElement.innerHTML = `${weather.temperature} &#176 <span> C</span>`;
    tempFeelsElement.innerHTML = `${weather.temperature_FeelsLike} &#176 <span> C</span>`;
    tempMinElement.innerHTML = `${weather.temperatureMin} &#176 <span> C</span>`;
    tempMaxElement.innerHTML = `${weather.temperatureMax} &#176 <span> C</span>`;
    humidityElement.innerHTML = `${weather.humidity} <span>&#37</span>`;
    descElement.innerHTML = weather.description;
    locationElement.innerHTML = `${weather.city}, ${weather.country}`;
}

**Comment by golovnevs** [comment]
Just repeating my message from slack here for history.

Good work! I think the main trouble just in button type, but I can't check it without your htlm file. Have you changed < button type="submit" > to < input type="button" >in your html file?
In JS file:
change 'submit' action to 'click' here:
document.getElementById('citysearch').addEventListener('submit', getWeather);

I see, that now the variable 'city_name' always empty. Because browser trying to assign value to the variable at once when page loading,  but input form 'citysearch' still empty. (code in js executing from top to bottom when loading)
You need to move string 'let city_name = document.getElementById("searchbar").value;'
into the function and in this case, variable will be assigned after pushing the search button  - when the input form already filled with the city name.

And one cosmetic thing - it's better to use camelCase, when you name the variables. I mean  citySearch, cityName, searchBar instead of city_search, cityname, searchBar

**Comment by marzieh67** [comment]
@golovnevs thank a lot. the search function is now working perfect in my local file, but after I merged it on github link (https://marzieh67.github.io/ReDI_Frontend_FinalProject/), it shows some error in console while getting data from api. that was strange :no_mouth:

**Comment by golovnevs** [comment]
Looks great! I like it! I see the errors too, but as I understood it is just a gitpages restrictions. Try to change protocol http to https in the request.

const api= `**http**://api.openweathermap....      change to  
const api= `**https:**//api.openweathermap.org/data/2.5/weather?q=${city}&appid=${key}&units=metric`;

**Comment by golovnevs** [comment]
Could you please check 'amcharts_weather_icons_1.0.0 folder'?  If you don't use it in app, it would be better add it to .gitignore file. (just in case info about .gitignore [https://www.freecodecamp.org/news/gitignore-what-is-it-and-how-to-add-to-repo/](https://www.freecodecamp.org/news/gitignore-what-is-it-and-how-to-add-to-repo/) )

**Comment by golovnevs** [comment]
Last thing about search - I see an error in the console after loading, we need to fix it. 

The reason is in eventlistener. How do you think, what's wrong here? :)
document.getElementById('citySearch').addEventListener('click', getWeather);

Part of the page, related to this eventlistener:
-------
<form id="citySearch">
      <div class="searchBox">
           City: <input type="text" id ="searchBar" placeholder="search for a city">
           <input type="button" value="Search">
          </div>
          </form> (edited)

**Comment by marzieh67** [comment]
> Last thing about search - I see an error in the console after loading, we need to fix it.
> 
> The reason is in eventlistener. How do you think, what's wrong here? :)
> 
> ### document.getElementById('citySearch').addEventListener('click', getWeather);
> Part of the page, related to this eventlistener:
> 
> ```
>   <div class="searchBox">
>        City: <input type="text" id ="searchBar" placeholder="search for a city">
>        <input type="button" value="Search">
>       </div>
>       </form> (edited) 
> ```

yes! thanks for reminding me about it. I corrected it and do not have this get error any more (still just local file)👍
